tokio/build.rs

```diff
@@ -19,4 +19,29 @@ fn main() {
             );
         }
     }
+
+    assert_platform_minimums();
+}
+
+/// Assert platform-minimum requirements for Tokio to work correctly. This might
+/// be feasible to do as a constant fn, once MSRC is bumped a bit more.
+fn assert_platform_minimums() {
+    use std::mem;
+    use std::sync::atomic::{AtomicIsize, AtomicUsize};
+
+    if mem::size_of::<usize>() < 4 {
+        panic!("Tokio only works correctly if usize is at least 4 bytes.");
+    }
```

Comment:
What about cross-compilation? Then the build script is not built in the target architecture. Another option is this:
```Rust
#[cfg_attr(target_pointer_width = "16", compile_error("Pointer width must be at least 32"))]
```

Reply:
Hm. While strictly speaking not exhaustive, that's probably good enough until an MSRV bump. If someone designs a 31-bit hobby platform today they only have themselves to blame. :D

Reply:
How about?

```rust
#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64", target_pointer_width = "128")))]
compile_error! {
    "Tokio require points widths to be 32, 64, or 128 bits wide"
}
```